client/signup/main.jsx

```diff
@@ -314,9 +315,23 @@ class Signup extends Component {
 
 	getRecordProps() {
 		const { signupDependencies } = this.props;
+		let theme = get( signupDependencies, 'selectedDesign.theme' );
+
+		if ( ! theme && signupDependencies.themeParameter ) {
+			theme = signupDependencies.themeParameter;
+		}
```

Comment:
Under which circumstances can `theme` be empty? In my tests is always defined.

Reply:
The method is called for all signup flows, not just the with-theme flow. Also, in the controller, we already check if it's null or not https://github.com/Automattic/wp-calypso/blob/11fcd29e2d573a5e54e7b7022149b343f45a0dd0/client/signup/controller.js#L290